modules/barcode/src/decoder/abs_decoder.hpp

```diff
@@ -0,0 +1,73 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+// Copyright (c) 2020-2021 darkliang wangberlinT Certseeds
+
+#ifndef __OPENCV_BARCODE_ABS_DECODER_HPP__
+#define __OPENCV_BARCODE_ABS_DECODER_HPP__
+
+#include <opencv2/barcode.hpp>
+
+namespace cv {
+namespace barcode {
+using std::string;
+using std::vector;
+constexpr static int BLACK = std::numeric_limits<uchar>::min();
+// WHITE elemental area is 0xff
+constexpr static int WHITE = std::numeric_limits<uchar>::max();
+
+
+struct Result
+{
+    std::string result;
+    BarcodeType format = BarcodeType::NONE;
+
+    Result() = default;
+
+    Result(std::string _result, BarcodeType _format)
+    {
+        result = std::move(_result);
+        format = _format;
+    }
+};
+
+struct Counter
+{
+    std::vector<int> pattern;
+    uint sum;
+
+    explicit Counter(vector<int> _pattern)
+    {
+        pattern = std::move(_pattern);
```

Comment:
const reference (to keep external interfaces clear) + assign operator without std::move?